src/componententity.cpp

```diff
@@ -192,6 +201,64 @@ void ComponentEntity::removeAllUnits()
     mPimpl->mUnits.clear();
 }
 
+bool ComponentEntity::hasUnits(const std::string &name) const
+{
+    auto result = mPimpl->findUnits(name);
+    return result != mPimpl->mUnits.end();
+}
+
+UnitsPtr ComponentEntity::getUnits(size_t index)
+{
+    return mPimpl->mUnits.at(index);
+}
+
+const UnitsPtr& ComponentEntity::getUnits(size_t index) const
+{
+    return mPimpl->mUnits.at(index);
+}
+
+UnitsPtr ComponentEntity::getUnits(const std::string &name)
+{
+    auto result = mPimpl->findUnits(name);
+    size_t index = result - mPimpl->mUnits.begin();
+    return mPimpl->mUnits.at(index);
```

Comment:
I think sometimes breaking up lines is helpful, but in this case (and possibly similar ones) the single-line version isn't that long.

It is however slightly counter-intuitive to me, since findUnits returns an iterator, that we don't just check it's valid and dereference, instead of computing an offset then using at!